From 26335a73f671b0f520e62547e6a5fd1b633fc06a Mon Sep 17 00:00:00 2001
From: Dan Williams <dan.j.williams@intel.com>
Date: Tue, 8 Mar 2022 21:28:05 -0800
Subject: [PATCH] build: Fix systemd unit directory detection

Upstream: https://lore.kernel.org/nvdimm/164680368507.2861995.11649228118524518218.stgit@dwillia2-desk3.amr.corp.intel.com

Older distributions (like CentOS Stream 8) define the systemd unit
directory variable in pkgconfig as ${systemdsystemunitdir}, instead of
${systemd_system_unit_dir}. Newer systemd pkgconfig uses both. Pick the
older more compatible name for use.

Fixes: 4e5faa1726d2 ("build: Add meson build infrastructure")
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 42e11aa..17ca65a 100644
--- a/meson.build
+++ b/meson.build
@@ -150,7 +150,7 @@ endif
 
 if get_option('systemd').enabled()
   systemd = dependency('systemd', required : true)
-  systemdunitdir = systemd.get_pkgconfig_variable('systemd_system_unit_dir')
+  systemdunitdir = systemd.get_pkgconfig_variable('systemdsystemunitdir')
   udev = dependency('udev', required : true)
   udevdir = udev.get_pkgconfig_variable('udevdir')
   udevrulesdir = udevdir / 'rules.d'
-- 
2.35.1

